test/generate-xml.sh

```diff
@@ -43,32 +43,40 @@ if [ ! -x $mbus_parse_hex ]; then
     exit 3
 fi
 
-for hexfile in "$directory"/*.hex;  do
-    if [ ! -f "$hexfile" ]; then
-        continue
-    fi
+generate_xml() {
+    directory="$1"
+    hexfile="$2"
+    mode="$3"
 
     filename=`basename $hexfile .hex`
 
+    if [ $mode = "normalized" ]; then
+        options="-n"
+        suffix=".norm.xml"
+    else
+        options=""
+        suffix=".xml"
+    fi
+
     # Parse hex file and write XML in file
-    $mbus_parse_hex "$hexfile" > "$directory/$filename.xml.new"
+    $mbus_parse_hex $options "$hexfile" > "$directory/$filename$suffix.new"
     result=$?
 
     # Check parsing result
     if [ $result -ne 0 ]; then
         echo "Unable to generate XML for $hexfile"
-        rm "$directory/$filename.xml.new"
-        continue
+        rm "$directory/$filename$suffix.new"
+        return 1
     fi
 
     # Compare old XML with new XML and write in file
-    diff -u "$directory/$filename.xml" "$directory/$filename.xml.new" 2> /dev/null > "$directory/$filename.dif"
+    diff -u "$directory/$filename$suffix" "$directory/$filename$suffix.new" 2> /dev/null > "$directory/$filename.dif"
```

Comment:
```suggestion
    diff -u "$directory/$filename$suffix" "$directory/$filename$suffix.new" 2> /dev/null > "$directory/$filename$suffix.dif"
```

I would like to run all tests at once, and see every failed test at once. Both not nromailized and normalized could fail. So it is better to add the suffix here. So there is no question about which test that failed.